src/Microsoft.Azure.WebJobs.Host/Retry/RetryManager.cs

```diff
@@ -0,0 +1,159 @@
+            {
+                LogWithId($"Function code returned retry settings: '{retryException.RetryResult.Format()}'");
+                await ExecuteWithRetriesInternalAsync(execute, executeParams, retryException.RetryResult, token);
+            }
+            catch (OperationCanceledException)
+            {
+                LogWithId("Retries were canceled");
+            }
+        }
+
+        private void RethrowExceptionIfNeeded(IEnumerable<FunctionResult> results, bool throwInvocationException)
+        {
+            foreach (var result in results)
+            {
+                if (!result.Succeeded && (result.Exception is RetryException || throwInvocationException))
+                {
+                    throw result.Exception;
+                }
+            }
+        }
+
+        private AsyncRetryPolicy GetRetryPolicy(RetryResult retry)
+        {
+            PolicyBuilder builder = Policy.Handle<FunctionInvocationException>();
+            if (retry.RetryCount == -1)
+            {
+                if (retry.ExponentialBackoff)
+                {
+                    return builder.WaitAndRetryForeverAsync(GetDurationExponential, OnRetryExponentialAsync);
+                }
+                else
+                {
+                    return builder.WaitAndRetryForeverAsync(GetDuration, OnRetryExponentialAsync);
+                }
+            }
+            else
+            {
+                if (retry.ExponentialBackoff)
+                {
+                    return builder.WaitAndRetryAsync(retry.RetryCount, GetDurationExponential, OnRetryAsync);
+                }
+                else
+                {
+                    return builder.WaitAndRetryAsync(retry.RetryCount, GetDuration, OnRetryAsync);
+                }
+            }
+        }
+
+        private TimeSpan GetDuration(int retryCount, Context context)
+        {
+            return ((RetryResult)context["retry"]).SleepDuration;
+        }
+
+        private TimeSpan GetDurationExponential(int retryCount, Context context)
+        {
+            return TimeSpan.FromSeconds(Math.Pow(2, retryCount));
+        }
+
+        private Task OnRetryAsync(Exception exception, TimeSpan timeSpan, int retryCount, Context context)
+        {
+            LogWithId($"New linar retry attempt, Exception: '{exception}', Timestamp: '{timeSpan}', RetryCount: {retryCount}");
+            context["retryCount"] = retryCount;
+            return Task.CompletedTask;
+        }
+
+        private Task OnRetryExponentialAsync(Exception exception, int retryCount, TimeSpan timeSpan, Context context)
+        {
+            LogWithId($"New exponential retry attempt, Exception: '{exception}', Timestamp: '{timeSpan}', RetryCount: {retryCount}");
+            context["retryCount"] = retryCount;
+            return Task.CompletedTask;
+        }
+
+        private void LogWithId(string message)
+        {
+            _logger.LogDebug($"Id: '{_id}', {message}");
```

Comment:
Why is this being called `id` and not `message`?

Reply:
because "id" is added to each log